<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GUI_ParamDeform</class>
 <widget class="QMainWindow" name="GUI_ParamDeform">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>782</width>
    <height>419</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Face Workshop</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Single Face Editing</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_8">
        <item row="0" column="0">
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <widget class="PhotoView" name="viewPic">
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>0</height>
            </size>
           </property>
          </widget>
          <widget class="QFrame" name="frame">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="0" column="0" colspan="2">
             <widget class="QTabWidget" name="tabProcessMethod">
              <property name="currentIndex">
               <number>1</number>
              </property>
              <widget class="QWidget" name="tab_3">
               <attribute name="title">
                <string>FreeDeform</string>
               </attribute>
               <layout class="QGridLayout" name="gridLayout_3">
                <item row="0" column="0">
                 <widget class="Widget_WarpParam" name="wWarpParam" native="true"/>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="tab_2">
               <attribute name="title">
                <string>Param</string>
               </attribute>
               <layout class="QGridLayout" name="gridLayout_2">
                <item row="0" column="1">
                 <widget class="QSlider" name="sldFatThin">
                  <property name="minimum">
                   <number>-50</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>Fatness</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>Eye Size</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QSlider" name="sldEyeSize">
                  <property name="minimum">
                   <number>-50</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_4">
                  <property name="text">
                   <string>Mouth Size</string>
                  </property>
                 </widget>
                </item>
                <item row="4" column="0">
                 <widget class="QLabel" name="label_de">
                  <property name="text">
                   <string>DeID</string>
                  </property>
                 </widget>
                </item>
                <item row="4" column="1">
                 <widget class="QSlider" name="sldDeidentify">
                  <property name="minimum">
                   <number>-50</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QSlider" name="sldParam7">
                  <property name="minimum">
                   <number>-50</number>
                  </property>
                  <property name="maximum">
                   <number>50</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="tab">
               <attribute name="title">
                <string>RawParam</string>
               </attribute>
               <layout class="QGridLayout" name="gridLayout_4">
                <item row="0" column="0" colspan="2">
                 <widget class="Widget_ModelParamEdit" name="wParamEdit" native="true"/>
                </item>
                <item row="1" column="1">
                 <widget class="QPushButton" name="pushButton">
                  <property name="text">
                   <string>Reset</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <spacer name="horizontalSpacer">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Image Warping </string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="cmbImgWarpAlg"/>
            </item>
            <item row="2" column="1">
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>201</width>
                <height>26</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="3" column="0" colspan="2">
             <widget class="QCheckBox" name="chkShowMarkPoint">
              <property name="text">
               <string>Show Mark Points</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QCheckBox" name="chkShowImage">
              <property name="text">
               <string>Show Image</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QCheckBox" name="chkDeform">
              <property name="text">
               <string>Deform</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <attribute name="title">
        <string>Multi Face Morphing</string>
       </attribute>
       <!--<layout class="QGridLayout" name="gridLayout_6">-->
        <!--<item row="0" column="0">-->
         <!--<widget class="Widget_FaceMorph" name="widgetFaceMorph" native="true"/>-->
        <!--</item>-->
       <!--</layout>-->
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>782</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoadImg"/>
    <addaction name="actCapFromCam"/>
    <addaction name="actionSaveOriImage"/>
    <addaction name="actionExport"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionZoom_In"/>
    <addaction name="actionZoom_out"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionLoadImg">
   <property name="text">
    <string>Load Image...</string>
   </property>
  </action>
  <action name="actCapFromCam">
   <property name="text">
    <string>Capture from camera...</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom In</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+=</string>
   </property>
  </action>
  <action name="actionZoom_out">
   <property name="text">
    <string>Zoom Out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="actionSaveOriImage">
   <property name="text">
    <string>Save Ori Image...</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="text">
    <string>Export Picture...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PhotoView</class>
   <extends>QGraphicsView</extends>
   <header>photoview.h</header>
   <slots>
    <signal>selectedPointUpdated(int)</signal>
    <signal>warpControlUpdated()</signal>
    <slot>setMirrorPoint(int)</slot>
    <slot>setImageVisibility(bool)</slot>
    <slot>setMarkPointVisibility(bool)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>Widget_ModelParamEdit</class>
   <extends>QWidget</extends>
   <header>widget_modelparamedit.h</header>
   <container>1</container>
   <slots>
    <signal>paramUpdated()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>Widget_WarpParam</class>
   <extends>QWidget</extends>
   <header>widget_warpparam.h</header>
   <container>1</container>
   <slots>
    <signal>updated()</signal>
   </slots>
  </customwidget>
  <!--<customwidget>-->
   <!--<class>Widget_FaceMorph</class>-->
   <!--<extends>QWidget</extends>-->
   <!--<header>widget_facemorph.h</header>-->
   <!--<container>1</container>-->
  <!--</customwidget>-->
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>chkShowMarkPoint</sender>
   <signal>toggled(bool)</signal>
   <receiver>viewPic</receiver>
   <slot>setMarkPointVisibility(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>486</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>206</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkShowImage</sender>
   <signal>toggled(bool)</signal>
   <receiver>viewPic</receiver>
   <slot>setImageVisibility(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>485</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabProcessMethod</sender>
   <signal>currentChanged(int)</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>751</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>386</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbImgWarpAlg</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>751</x>
     <y>275</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>155</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wWarpParam</sender>
   <signal>updated()</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>740</x>
     <y>208</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldFatThin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>547</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>521</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wParamEdit</sender>
   <signal>paramUpdated()</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>528</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>viewPic</sender>
   <signal>warpControlUpdated()</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>318</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>272</x>
     <y>44</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldDeidentify</sender>
   <signal>valueChanged(int)</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>530</x>
     <y>195</y>
    </hint>
    <hint type="destinationlabel">
     <x>421</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldEyeSize</sender>
   <signal>valueChanged(int)</signal>
   <receiver>GUI_ParamDeform</receiver>
   <slot>updatePic()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>585</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>734</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updatePic()</slot>
 </slots>
</ui>
